site/content/docs/main/config/gateway-api.md

```diff
@@ -0,0 +1,211 @@
+
+In **dynamic** provisioning, the platform operator first deploys Contour's Gateway provisioner. Then, the platform operator defines a `Gateway` resource, and the provisioner automatically deploys a Contour instance that corresponds to the `Gateway's` configuration and will process that `Gateway` and its routes.
+
+Static provisioning makes sense for users who: 
+- prefer the traditional model of deploying Contour
+- have only a single Gateway
+- want to use just the standard listener ports (80/443)
+- have highly customized YAML for deploying Contour.
+
+Dynamic provisioning makes sense for users who:
+- have many Gateways
+- want to use additional listener ports
+- prefer a simple declarative API for provisioning Contour instances
+- want a fully conformant Gateway API implementation
+
+### Static Provisioning
+
+To statically provision Contour with Gateway API enabled:
+
+1. Install the [Gateway API experimental channel][3].
+1. Create a GatewayClass, with a controller name of `projectcontour.io/gateway-controller`.
+1. Create a Gateway using the above GatewayClass.
+1. In the Contour config file, add a reference to the above Gateway via `gateway.gatewayRef` (see https://projectcontour.io/docs/1.25/configuration/#gateway-configuration)
+1. Install Contour using the above config file.
+
+Contour provides an example manifest for this at https://projectcontour.io/quickstart/contour-gateway.yaml.
+
+### Dynamic Provisioning
+
+To dynamically provision Contour with Gateway API enabled:
+
+1. Install the [Contour Gateway Provisioner][9], which includes the Gateway API experimental channel.
+1. Create a GatewayClass, with a controller name of `projectcontour.io/gateway-controller`.
+1. Create a Gateway using the above GatewayClass.
+
+The Contour Gateway Provisioner will deploy an instance of Contour in the Gateway's namespace implementing the Gateway spec.
+
+## Gateway Listeners
+
+Each unique Gateway Listener port requires the Envoy service to expose that port, and to map it to an underlying port in the Envoy daemonset/deployment that Envoy is configured to listen on.
+For example, the following Gateway Listener configuration (abridged) requires service ports of 80 and 443, mapped to underlying container ports 8080 and 8443:
+
+```yaml
+listeners:
+- name: http
+  protocol: HTTP
+  port: 80
+- name: https
+  protocol: HTTPS
+  port: 443
+```
+
+In dynamic provisioning, the Contour Gateway Provisioner will continuously ensure that the Envoy service and daemonset/deployment are kept in sync with the Gateway Listener configuration.
+In static provisioning, it is up to the platform operator to keep the Envoy resources in sync with the Gateway Listeners.
+
+To get from the Gateway Listener port to the port that Envoy will be configured to listen on, i.e. the container port:
+- add 8000 to the Listener port number
+- if the result is greater than 65535, subtract 65535
+- if the result is less than or equal to 1023, add 1023.
```

Comment:
nit: maybe mention this could still have conflicts, and to check on GW status?

Reply:
https://github.com/projectcontour/contour/pull/5433/commits/d738d078046871a6a9cbb222bd1b34332dc06c0a